File: solutions/javascript/RemoveDuplicatesfromSortedArrayII/RemoveDuplicatesfromSortedArrayII.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    if (nums.length < 3) {
        return nums.length;
    }
    var n = 0;
    for (var i = 2; i < nums.length; i++) {
        if (nums[i] > nums[n]) {
            nums[n+2] = nums[i];
            n++;
        }
    }
    return n+2;
};

File: solutions/java/RemoveDuplicatesfromSortedArrayII/RemoveDuplicatesfromSortedArrayII.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length < 3) {
            return nums.length;
        }
        int n = 0;
        for (int i = 2; i < nums.length; i++) {
            if (nums[i] != nums[n]) {
                n++;
                nums[n+1] = nums[i];
            }
        }
        return n + 2;
    }
}